### PR TITLE
Add simple support of Mariner OS distribution to installdependencies.sh

### PR DESCRIPTION
**Pull request by EzzhevNikita**
This PR is adding simple support of Mariner OS distribution to installdependencies.sh script.
To execute agent on Mariner OS distribution requires additional installation of _icu_ package, so it was added step for installing this lib for the case when Mariner OS was detected as host OS.

Tested changes on a local installation of Mariner OS. It was tested case of agent setup on a clean OS. With updated installdependencies.sh script agent was successfully set up and was able to connect to the ADO organization. 
Instruction for an installation could be found [here](https://github.com/microsoft/CBL-Mariner/blob/1.0/toolkit/docs/quick_start/quickstart.md#prepare-vm)
ISO image of OS is [here](https://aka.ms/mariner-1.0-x86_64-iso)

### CODE REVIEW

File: src/Misc/layoutbin/installdependencies.sh [hunk 1]
```diff
@@ -257,6 +256,28 @@ then
...
+        then
+            echo "The current OS is Mariner based"
+            echo "--------Mariner Version--------"
+            cat /etc/mariner-release
+            echo "------------------------------"
+
+            command -v yum
+            if [ $? -eq 0 ]
+                then
+                yum install -y icu
```
Comment (DaniilShmelev) [inline src/Misc/layoutbin/installdependencies.sh:269]: Is this the only dependency we need to install on this OS?
As I can see on other OSs we install openssl/lttng-ust/...

Comment (EzzhevNikita) [reply src/Misc/layoutbin/installdependencies.sh:269]: Yes, it is enough for Mariner OS. I have tested it and didn't mention any other dependencies that should be installed.

File: src/Misc/layoutbin/installdependencies.sh [hunk 2]
```diff
@@ -257,6 +256,28 @@ then
                 print_errormessage
                 exit 1
             fi
+        elif [ -e /etc/mariner-release ]
+        then
+            echo "The current OS is Mariner based"
```
Comment (DaniilShmelev) [inline src/Misc/layoutbin/installdependencies.sh:261]: On [line 10](https://github.com/microsoft/azure-pipelines-agent/pull/3698/files#R10), there's a small description of how different OSs are detected. Could you please also add a note about Mariner OS there?

**Review by DaniilShmelev — APPROVED** [review]
LGTM but take a look at the comment

**Review by anatolybolshakov — APPROVED** [review]

**Comment by anatolybolshakov** [comment]
@EzzhevNikita seems like checks failed - could you please take a look?

**Comment by EzzhevNikita** [comment]
> @EzzhevNikita seems like checks failed - could you please take a look?

Failures caused by outdated dotnet-install script, I created PR to address this problem.

**Pull request merged**